pyproject.toml

```diff
@@ -64,10 +66,12 @@ test = [
     "flow.record[avro]",
     "flow.record[elastic]",
     "flow.record[xlsx]",
-    "duckdb; platform_python_implementation != 'PyPy' and python_version < '3.12'", # duckdb
-    "pytz; platform_python_implementation != 'PyPy' and python_version < '3.12'", # duckdb
+    "duckdb; platform_python_implementation != 'PyPy' and python_version < '3.12'", # Don't install duckdb on PyPy
+    "pytz; platform_python_implementation != 'PyPy' and python_version < '3.12'", # Don't install duckdb on PyPy
     "tqdm",
     "structlog",
+    "pytest",
+    "ruff==0.9.2",
 ]
```

Comment:
Will you remove the `test` extra and update the `tox.ini` to use `dependency_groups`?